openfaas/faas-cli
# [[#779](https://github.com/openfaas/faas/issues/779)] Add `latest` Tag to New Functions

Add `latest` tag to new functions created by OpenFaaS CLI tool.

## Description


Append `latest` tag to all newly created functions. This fixes a bug
where if you have multiple tags for an image and push without a tag, all
versions are pushed to the remote side. Now, only the current tag will
be pushed.

Signed-off-by: James Smith <jamessmithsmusic@gmail.com>

## Motivation and Context



References [[#779](https://github.com/openfaas/faas/issues/779)] in [openfaas/faas](https://github.com/openfaas/faas).

## How Has This Been Tested?




Existing tests were ran successfully.

Verified existing functions without a tag are not broken by :

- Creating new function
- removing supplied tag `latest`
- Build / Push function
- Update tag to `v0.0.1`
- Build / Push function
- verify that only tag `v0.0.1` is updated
- Remove tag
- verify that both tags were updated

Verified new functionality by :

- Creating new function
  - template tag used is `latest`
- Build / Push function
- Update tag to `v0.0.1`
- Build / Push function
- verify that only tag `v0.0.1` is updated
- Update tag to `latest`
- verify that only tag `latest` is updated

## Types of changes


## Checklist:


  - Modified current tests to satisfy.
